pybamm/solvers/klu_sparse_solver.py

```diff
@@ -0,0 +1,169 @@
+#
+# Solver class using sundials with the KLU sparse linear solver
+#
+import pybamm
+import numpy as np
+import scipy.sparse as sparse
+
+import importlib
+
+klu_spec = importlib.util.find_spec("klu")
+if klu_spec is not None:
+    klu = importlib.util.module_from_spec(klu_spec)
+    klu_spec.loader.exec_module(klu)
+
+
+def have_klu():
+    return klu_spec is None
+
+
+class KLU(pybamm.DaeSolver):
+    """Solve a discretised model, using sundials with the KLU sparse linear solver.
+
+     Parameters
+    ----------
+    rtol : float, optional
+        The relative tolerance for the solver (default is 1e-6).
+    atol : float, optional
+        The absolute tolerance for the solver (default is 1e-6).
+    root_method : str, optional
+        The method to use to find initial conditions (default is "lm")
+    root_tol : float, optional
+        The tolerance for the initial-condition solver (default is 1e-8).
+    max_steps: int, optional
+        The maximum number of steps the solver will take before terminating
+        (default is 1000).
+    """
+
+    def __init__(
+        self, rtol=1e-6, atol=1e-6, root_method="lm", root_tol=1e-6, max_steps=1000
+    ):
+
+        if klu_spec is None:
+            raise ImportError("KLU is not installed")
+
+        super().__init__("ida", rtol, atol, root_method, root_tol, max_steps)
+
+    def integrate(self, residuals, y0, t_eval, events, mass_matrix, jacobian):
```

Comment:
does this work if events and jacobian are None? If not, could you raise an exception when they arent passed